sdk/keyvault/Azure.Security.KeyVault.Keys/src/CreateKeyOptions.cs

```diff
@@ -34,6 +41,6 @@ public class KeyCreateOptions
         /// <summary>
         /// A dictionary of tags with specific metadata about the key.
         /// </summary>
-        public IDictionary<string, string> Tags { get; private set; } = new Dictionary<string, string>();
+        public IDictionary<string, string> Tags { get; } = new Dictionary<string, string>();
```

Comment:
Looking at how this is serialized I think it will cause the request to contain "tags" : { }. Not sure if this might have unintended consequences

Reply:
The `KeyRequestParameters` made a copy, and did check the `Count`. That said, even making a copy if empty doesn't make sense, though I'm not even sure why the code is making a copy now. Partial immutability in multi-threaded scenarios, perhaps?

I also fixed `KeyOperations` similarly. Makes me wonder if we're not providing enough coverage to make sure we're not zero'ing out tags and/or ops. I'll open a bug.

Reply:
#8167 opened to track adding coverage.